### PR TITLE
rename

### PR DESCRIPTION
**Pull request by blagoev**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -82,7 +82,7 @@ For a complete documentation go to [Realm Flutter and Dart SDK Docs](https://doc
 
 ## Limitations
 
-* This version of Realm Flutter and Dart SDK allows working with a local only (on device) Realm database in Flutter and Dart desktop. Realm Sync functionality is not implemented.
+* This version of Realm Flutter and Dart SDK allows working with a local only (on device) Realm database in Flutter and Dart desktop. Atlas Device Sync functionality is not implemented.
```
Comment (nirinchev) [inline README.md:85]: Should we still list that limitation? It should be gone with the next release, so might as well just delete it from the readme.

File: lib/src/app.dart [hunk 2]
```diff
@@ -38,7 +38,7 @@ class AppConfiguration {
   /// directory exists.
   final Directory baseFilePath;
 
-  /// The [baseUrl] is the [Uri] used to reach the MongoDB Realm server.
+  /// The [baseUrl] is the [Uri] used to reach the MongoDB Atlas.
```
Comment (nirinchev) [inline lib/src/app.dart:41]:
```suggestion
  /// The [baseUrl] is the [Uri] used to reach Atlas App Services.
```

Technically, those are different urls today - MongoDB Atlas is at `cloud.mongodb.com` and AAS is at `realm.mongodb.com`. I imagine the second will change, but stay different. So users shouldn't type in the Atlas url as the `baseUrl`.

File: lib/src/app.dart [hunk 3]
```diff
@@ -80,7 +80,7 @@ class AppConfiguration {
   /// The [HttpClient] that will be used for HTTP requests during authentication.
   ///
   /// You can use this to override the default http client handler and configure settings like proxies,
-  /// client certificates, and cookies. While these are not required to connect to MongoDB Realm under
+  /// client certificates, and cookies. While these are not required to connect to MongoDB Atlas under
```
Comment (nirinchev) [inline lib/src/app.dart:83]:
```suggestion
  /// client certificates, and cookies. While these are not required to connect to Atlas App Services under
```

We don't connect to atlas directly - it's always via AAS, even when using remote mongo and functions.

File: lib/src/cli/deployapps/baas_client.dart [hunk 4]
```diff
@@ -67,7 +67,7 @@ class BaasClient {
         _headers = <String, String>{'Accept': 'application/json'},
         _appSuffix = '-$_clusterName';
 
-  /// A client that imports apps in a MongoDB Realm docker image. See https://github.com/realm/ci/tree/master/realm/docker/mongodb-realm
+  /// A client that imports apps in a MongoDB Atlas docker image. See https://github.com/realm/ci/tree/master/realm/docker/mongodb-realm
```
Comment (nirinchev) [inline lib/src/cli/deployapps/baas_client.dart:70]:
```suggestion
  /// A client that imports apps in an Atlas App Services docker image. See https://github.com/realm/ci/tree/master/realm/docker/mongodb-realm
```

Similarly here - we don't have any Atlas docker images and the AAS docker image is running a local MDB server, not Atlas.

File: lib/src/cli/deployapps/baas_client.dart [hunk 5]
```diff
@@ -83,7 +83,7 @@ class BaasClient {
     return result;
   }
 
-  /// A client that imports apps to an MongoDB Realm environment (typically realm-dev or realm-qa).
+  /// A client that imports apps to a MongoDB Atlas environment (typically realm-dev or realm-qa).
```
Comment (nirinchev) [inline lib/src/cli/deployapps/baas_client.dart:86]:
```suggestion
  /// A client that imports apps to an Atlas App Services environment (typically realm-dev or realm-qa).
```

Not sure what these environments will change to, but probably not `realm-dev` and `realm-qa`. I asked on Slack.

File: lib/src/cli/deployapps/deployapps_command.dart [hunk 6]
```diff
@@ -26,7 +26,7 @@ import 'baas_client.dart';
 
 class DeployAppsCommand extends Command<void> {
   @override
-  final String description = 'Deploys test applications to a MongoDB Realm server.';
+  final String description = 'Deploys test applications to MongoDB Atlas.';
```
Comment (nirinchev) [inline lib/src/cli/deployapps/deployapps_command.dart:29]:
```suggestion
  final String description = 'Deploys test applications to Atlas App Services.';
```

File: lib/src/cli/deployapps/options.dart [hunk 7]
```diff
@@ -22,7 +22,7 @@ part 'options.g.dart';
 
 @CliOptions()
 class Options {
-  @CliOption(help: 'Url for MongoDB Realm.', defaultsTo: 'http://localhost:9090')
+  @CliOption(help: 'Url for MongoDB Atlas.', defaultsTo: 'http://localhost:9090')
```
Comment (nirinchev) [inline lib/src/cli/deployapps/options.dart:25]:
```suggestion
  @CliOption(help: 'Url for Atlas App Services.', defaultsTo: 'http://localhost:9090')
```

File: lib/src/session.dart [hunk 8]
```diff
@@ -71,7 +71,7 @@ class Session {
 
 /// The current state of a [Session] object
 enum SessionState {
-  /// The session is connected to the MongoDB Realm server and is actively transferring data.
+  /// The session is connected to  MongoDB Atlas and is actively transferring data.
```
Comment (nirinchev) [inline lib/src/session.dart:74]:
```suggestion
  /// The session is connected to Atlas Device Sync and is actively transferring data.
```

File: lib/src/session.dart [hunk 9]
```diff
@@ -80,13 +80,13 @@ enum SessionState {
 
 /// The current connection state of a [Session] object
 enum ConnectionState {
-  /// The session is disconnected from the MongoDB Realm server.
+  /// The session is disconnected from MongoDB Atlas.
```
Comment (nirinchev) [inline lib/src/session.dart:83]:
```suggestion
  /// The session is disconnected from Atlas Device Sync.
```

File: lib/src/session.dart [hunk 10]
```diff
@@ -80,13 +80,13 @@ enum SessionState {
...
   disconnected,
 
-  /// The session is connecting to the MongoDB Realm server.
+  /// The session is connecting to MongoDB Atlas.
```
Comment (nirinchev) [inline lib/src/session.dart:86]:
```suggestion
  /// The session is connecting to Atlas Device Sync.
```

File: lib/src/session.dart [hunk 11]
```diff
@@ -80,13 +80,13 @@ enum SessionState {
...
   connecting,
 
-  /// The session is connected to the MongoDB Realm server.
+  /// The session is connected to MongoDB Atlas.
```
Comment (nirinchev) [inline lib/src/session.dart:89]:
```suggestion
  /// The session is connected to Atlas Device Sync.
```

File: lib/src/user.dart [hunk 12]
```diff
@@ -46,7 +46,7 @@ class User {
     return realmCore.userGetState(this);
   }
 
-  /// Get this [User]'s id on MongoDB Realm.
+  /// Get this [User]'s id on MongoDB Atlas.
```
Comment (nirinchev) [inline lib/src/user.dart:49]:
```suggestion
  /// Get this [User]'s id in Atlas App Services.
```

File: lib/src/user.dart [hunk 13]
```diff
@@ -96,7 +96,7 @@ class User {
   ///
   /// Linking a user with more credentials, mean the user can login either of these credentials. It also makes it possible to "upgrade" an anonymous user
   /// by linking it with e.g. Email/Password credentials.
-  /// *Note: It is not possible to link two existing users of MongoDB Realm. The provided credentials must not have been used by another user.*
+  /// *Note: It is not possible to link two existing users of MongoDB Atlas. The provided credentials must not have been used by another user.*
```
Comment (nirinchev) [inline lib/src/user.dart:99]:
```suggestion
  /// *Note: It is not possible to link two existing users of Atlas App Services. The provided credentials must not have been used by another user.*
```

File: lib/src/user.dart [hunk 14]
```diff
@@ -129,7 +129,7 @@ enum UserState {
   /// The user is logged out. Call [App.logIn] to log the user back in.
   loggedOut,
 
-  /// The user is logged in, and any Realms associated with it are synchronizing with MongoDB Realm.
+  /// The user is logged in, and any Realms associated with it are synchronizing with MongoDB Atlas.
```
Comment (nirinchev) [inline lib/src/user.dart:132]:
```suggestion
  /// The user is logged in, and any Realms associated with it are synchronizing with Atlas Device Sync.
```

File: src/sync_session.h [hunk 15]
```diff
@@ -76,7 +76,7 @@ RLM_API uint64_t realm_dart_sync_session_register_connection_state_change_callba
  *  @param errorCode Error code of the error that to be simulated.
  *  @param isFatal >If set to `true` the error will be marked as fatal.
  *
- *  Use this method to test your error handling code without connecting to a MongoDB Realm Server.
+ *  Use this method to test your error handling code without connecting to MongoDB Atlas.
```
Comment (nirinchev) [inline src/sync_session.h:79]:
```suggestion
 *  Use this method to test your error handling code without connecting to Atlas Device Sync.
```

**Review by nirinchev — CHANGES_REQUESTED** [review]
I have suggested renaming most of the places where we use MongoDB Atlas to Atlas App Services. The reason is that we should avoid confusion between the two and be explicit that we're not connecting to Atlas directly (i.e. there's no need to whitelist IPs, or provision users for the client apps).

**Comment by nielsenko** [comment]
It seems @nirinchev is on top of this review, I will remove myself.

File: README.md (hunk 1 above)
Comment (blagoev) [reply README.md:85]: No we should have this limitation any more. Will fix

**Pull request merged**